### PR TITLE
fix #15143, fix #15959: reenable geokrety search using trackingcodes

### LINKED ISSUE #15143: GeoKrety: search trackables with secret trackingCode not working (any more?)
While refactoring trackable logging I played around with GeoKrety for the first time.

I noticed that searching a GeoKrety using its secret tracking code does not work in c:geo. Is this a bug or is this intended behaviour?

I found code in c:geo indicating that at some point in the past such a search would work. 
It seems c:geo tries to get the geocode for a trackable by calling the following URL: `https://api.geokrety.org/nr2id/{secretTrackingCode}`. However, this URL returns HTTP-404
(Reference to c:geo code: `GeokretyConnector.getGeocodeFromTrackingCode`)

### LINKED ISSUE #15959: Spoiler images no longer loaded
Spoiler images are no longer loaded. 

Example cache: GC2JVEH. This cache has 3 spoiler images, but they are no longer loaded.
This error makes the following test case fail: [cgeo.geocaching.CgeoApplicationTest.testSearchByGeocodeBasis](https://ci.cgeo.org/job/cgeo-CI_PR-build/9151/testReport/junit/cgeo.geocaching/CgeoApplicationTest/testSearchByGeocodeBasis/)

This seems to be caused by a HTML change from geocaching.com

### PR DESCRIPTION
**Pull request by eddiemuc**
fix #15143: reenable geokrety search using trackingcodes

### CODE REVIEW

File: main/src/main/java/cgeo/geocaching/connector/trackable/GeokretyConnector.java [hunk 1]
```diff
@@ -296,12 +299,30 @@ public String getTrackableTrackingCodeFromUrl(@NonNull final String url) {
...
+        //Example response:
+        //<gkxml version="1.0" date="2024-08-07 19:02:22" date_Iso8601="2024-08-07T19:02:22+00:00">
+        //<geokrety>
+        //<geokret id="66184" type="1" collectible="false" owner_id="26422" ownername="kumy" holder_id="26422" holdername="" dist="36291" date="2017-12-13" places="12" image="GK10288_6503b68749e8a">
+        //<![CDATA[ Super Picsou Geant #200 ]]>
+        //</geokret>
+        //</geokrety>
+        //</gkxml>
+
+        final String response = Network.getResponseData(Network.getRequest("https://staging.geokrety.org/api/v1/export2?tracking_code=" + trackingCode));
```
Comment (kumy) [inline main/src/main/java/cgeo/geocaching/connector/trackable/GeokretyConnector.java:312]: SUing the staging env is not a good idea as it's not prod !

And I thought there was some variable for the url: `URLPROXY`

Comment (eddiemuc) [reply main/src/main/java/cgeo/geocaching/connector/trackable/GeokretyConnector.java]: Changed to URL (which points to https://geokrety.org). I also tried URLPROXY (points to https://api.geokrety.org), but that gave me HTTP404

**Comment by eddiemuc** [comment]
Retest this please

**Comment by eddiemuc** [comment]
Seems like tests are really failing due to this PR. I have to look into it

**Comment by bekuno** [comment]
Retest this please

File: main/src/main/java/cgeo/geocaching/connector/trackable/GeokretyConnector.java (hunk 1 above)
Comment (kumy) [reply main/src/main/java/cgeo/geocaching/connector/trackable/GeokretyConnector.java]: I don't remember what happen to https://api.geokrety.org/ during the transition GKv2. Maybe it has been abandoned. Need to check

Comment (kumy) [reply main/src/main/java/cgeo/geocaching/connector/trackable/GeokretyConnector.java]: The `api.geokrety.org` url is not abandoned, it does only redirects. Some sample usages:
- `curl https://api.geokrety.org/gk/GKb65c/ -L`
- `curl https://api.geokrety.org/gk/GKb65c/details/ -L`
- `curl https://api.geokrety.org/wpt/GC5brqk/ -L`
- `curl https://api.geokrety.org/wpt/GC5brqk/details/ -L`

However it does not contain endpoint for searching by TC

You can check the nginx config here https://github.com/geokrety/geokrety-nginx/blob/main/files/etc/nginx/templates/default.conf.template#L183-L280

Comment (kumy) [reply main/src/main/java/cgeo/geocaching/connector/trackable/GeokretyConnector.java]: I remember now, this api is a relic of the old GeoKretyMap Api where I didn't had access to TC.
Better not using it anyway...

**Comment by eddiemuc** [comment]
retest this please

**Comment by eddiemuc** [comment]
retest this please

**Comment by eddiemuc** [comment]
retest this please

**Comment by eddiemuc** [comment]
CI is in a bad state and unable to complete a test run w/o crashing. I executed test cases locally, they are green. Thus I will merge now

**Comment by bekuno** [comment]
Beside the further existing error with https://api.geokrety.org, see https://github.com/cgeo/cgeo/issues/15143#issuecomment-2267650560, I got now 2 other geokrety releted errors on local tests. I will create a separate Issue.

**Pull request merged**